winml/lib/Api/NumericData.cpp

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include "pch.h"
+
+#include "impl/NumericData.h"
+#include "VectorBackedBuffer.h"
+#include "robuffer.h"
+#include "winrt/Windows.Storage.Streams.h"
+#include "DisjointBufferHelpers.h"
+
+namespace _winml {
+
+std::shared_ptr<_winml::idata> numeric_data::create(
+  size_t num_elements,
+  size_t element_size_in_bytes,
+  wfc::IIterable<wss::IBuffer> const& buffers) {
+  return std::make_shared<numeric_data>(num_elements, element_size_in_bytes, buffers);
+}
+
+numeric_data::numeric_data(
+  size_t num_elements, size_t element_size_in_bytes, wfc::IIterable<wss::IBuffer> const& buffers) :
+  num_elements_(num_elements),
+  element_size_in_bytes_(element_size_in_bytes),
+  combined_buffer_(nullptr),
+  buffers_() {
+  if (buffers != nullptr) {
+    buffers_ = { begin(buffers), end(buffers) };
```

Comment:
Is this a Microsoft extension? I don't see an overload to assign a pair of iterators ([`operator=` on cppreference](https://en.cppreference.com/w/cpp/container/vector/operator%3D)), I though you had to call [`assign`](https://en.cppreference.com/w/cpp/container/vector/assign)

Reply:
I believe this will call the 
    vector& operator=(initializer_list<_Ty> _Ilist)?

Reply:
:+1: , it constructs a new vector and `=` it. The brackets had confused me a bit.